File: STRINGS/1632/main.dart
import 'dart:io';
import 'dart:math';

num getNumVariations(String s) {
  return pow(3, RegExp('[AEIOS]', caseSensitive: false).allMatches(s).length) * pow(2, RegExp('[^AEIOS]', caseSensitive: false).allMatches(s).length);
}

void main() {
  int t = int.parse(stdin.readLineSync()!);
  for (int test = 0; test < t; test++) {
    String s = stdin.readLineSync()!.trim();
    num numVariations = getNumVariations(s);
    print(numVariations);
  }
}

File: STRINGS/1632/main.js
function getNumVariations(s) {
    return Math.pow(3, (s.match(/[AEIOS]/ig) || []).length) * Math.pow(2, (s.match(/[^AEIOS]/ig) || []).length);
}

let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
let lines = input.split('\n');
let t = parseInt(lines[0]);
for (let i = 1; i <= t; i++) {
    let s = lines[i].trim();
    let numVariations = getNumVariations(s);
    console.log(numVariations);
}
